src/main/java/gov/cabinetoffice/gap/adminbackend/services/GrantAdvertService.java

```diff
@@ -89,10 +86,22 @@ public GrantAdvert create(Integer grantSchemeId, Integer grantAdminId, String na
                     "User " + grantAdminId + " is unable to access scheme with id " + scheme.getId());
         }
         final Integer version = featureFlagsProperties.isNewMandatoryQuestionsEnabled() ? 2 : 1;
-        final GrantAdvert grantAdvert = GrantAdvert.builder().grantAdvertName(name).scheme(scheme).createdBy(grantAdmin)
-                .created(Instant.now()).lastUpdatedBy(grantAdmin).lastUpdated(Instant.now())
-                .status(GrantAdvertStatus.DRAFT).version(version).build();
-        return this.grantAdvertRepository.save(grantAdvert);
+        final boolean doesAdvertExist = grantAdvertRepository.findBySchemeId(grantSchemeId).isPresent();
+
+        if (!doesAdvertExist) {
+            final GrantAdvert grantAdvert = GrantAdvert.builder().grantAdvertName(name).scheme(scheme)
+                    .createdBy(grantAdmin).created(Instant.now()).lastUpdatedBy(grantAdmin).lastUpdated(Instant.now())
+                    .status(GrantAdvertStatus.DRAFT).version(version).build();
+            return this.grantAdvertRepository.save(grantAdvert);
+        }
```

Comment:
don't need this else. if !doesAdvertExist will return and terminate this function